saleor/plugins/tests/sample_plugins.py

```diff
@@ -366,6 +367,24 @@ def payment_method_initialize_tokenization(self, request_data, previous_value):
     def payment_method_process_tokenization(self, request_data, previous_value):
         return previous_value
 
+    def get_shipping_methods_for_checkout(
+        self, checkout: "Checkout", previous_value: Any
+    ) -> list["ShippingMethodData"]:
+        different_currency = "EUR"
+        assert checkout.currency != different_currency
```

Comment:
@IKarbowiak this can be a nice addition to my statement about tests being explicit. So, we dont have to rely on "some nexted fixtures" being in "given" state we want. We also dont have to re-introduce them in every test, but we can assert fixtures being in valid state. If fixture changes, we fail fast (we know that "given" is broken before we check "then")

Reply:
BTW, should we do such assertion in the `SamplePlugin`?

Assertion should in specific test unit IMO.

Reply:
`assert` is not unit-test specific. You can find multiple assertions in the core code. 
In that case, I can't make `assert` inside the test, as sample plugin returns shipping methods with different currencies, but it is not available anymore for Saleor (PluginsManager, will filter it out), so only the `assert` here can confirm that  sample plugin returns the shipping methods different currency than checkout's currency (where single record has the same currency as the checkout's currency)